contributing.md

```diff
@@ -0,0 +1,141 @@
+
+Before creating enhancement suggestions, please **perform a [cursory search](https://github.com/search?q=is%3Aissue+repo%3Ametafacture%2Fmetafacture-playground)** to see if the enhancement has already been suggested. If it has, add a comment to the existing issue instead of opening a new one. When you are creating an enhancement suggestion, please include as many details as possible:
+
+#### How Do I Submit An Enhancement Suggestion?
+
+Enhancement suggestions are tracked as [GitHub issues](https://guides.github.com/features/issues/). Create an issue on the repository and provide the following information:
+
+* **Use a clear and descriptive title** for the issue to identify the suggestion.
+* **Provide a step-by-step description of the suggested enhancement** in as many details as possible.
+* **Provide specific examples to demonstrate the steps**. Include copy/pasteable snippets which you use in those examples, as Markdown code blocks.
+* **Describe the current behavior** and **explain which behavior you expected to see instead** and why.
+* **Explain why this enhancement would be useful** to most users.
+* **List some other applications where this enhancement exists.**
+* **Specify the name and version of the browser you're using.**
+
+### Pull Requests
+
+Please follow these steps to have your contribution considered by the maintainers:
+
+After you submit your pull request, verify that all [Github Actions](https://docs.github.com/en/actions) are passing.
+<details><summary>What if the actions are failing?</summary>If a action is failing, and you believe that the failure is unrelated to your change, please leave a comment on the pull request explaining why you believe the failure is unrelated. If we conclude that the failure was a false positive, then we will open an issue to track that problem with our actions</details>
+
+The reviewer(s) may ask you to complete additional design work, tests, or other changes before your pull request can be ultimately accepted.
+
+## Conventions
+
+### Git
+Git commits should be as granular as possible. When working on a fix for issue X, we try not to add other things we notice (typos, formatting, refactorings, etc.) to the same commit. We don’t use the GitHub shortcuts for closing issues from commits (like fixes #111), since in our process the issue is not solved by the commit but by the reviewed change after it’s deployed to production.
+
+### Commit Messages
+
+* Use the present tense ("Add feature" not "Added feature")
+* Use the imperative mood ("Move cursor to..." not "Moves cursor to...")
+* Use short lines (max 72 chars), and either just one line, or one line, a blank line, and one or more paragraphs.
+* Reference issues after the first line ("See #23")
+
```

Comment:
If I understand this correctly then the initial commit for of this PR (f09eafd69bd2b6eca1de4b63903c2e8c72d69d9e) does not follow that rule, right?

Reply:
If we use the short form (`#14`) I think it's OK to put it in the same line. I prefer full links (`https://github.com/metafacture/metafacture-playground/issues/14`), to make the git history  self-contained outside of GitHub, and these should always be in a separate line.

Reply:
I use the full links only when necessary, i.e. when referring to another repo. The short form implies the full link when it's the same repo, but I admit that erroneously using the short form (implying the false full link) when speaking about an other repo would be more unlikely if using the full form constantly.

Reply:
What I like about full links is that I can click them when looking at the history locally, with `git log` or `gitk`.

Reply:
I am sceptical about full links because full links might be a problem if/when we migrate from GitHub to another service. When you only have short links, they could remain intact and point to the migrated issue on the new platform. As the issues on GitHub probably won't go away, this might be a non-problem.

Reply:
It's true that clicking is easier but it's easier to _write_ just the short reference and it makes the messages not look so overcrowded, thus making the parsing by humans easier. I am ok with it,though, just saying.